instrumentation-core/spring/spring-webflux-5.0/src/main/java/io/opentelemetry/instrumentation/springwebflux/client/WebClientTracingFilter.java

```diff
@@ -52,29 +49,21 @@ public static void addFilter(
 
   @Override
   public Mono<ClientResponse> filter(final ClientRequest request, final ExchangeFunction next) {
-    final Span span =
-        tracer.spanBuilder(DECORATE.spanNameForRequest(request)).setSpanKind(CLIENT).startSpan();
+    final Span span = DECORATE.getOrCreateSpan(request, tracer);
     DECORATE.afterStart(span);
 
     try (final Scope scope = TRACER.withSpan(span)) {
       final ClientRequest mutatedRequest =
           ClientRequest.from(request)
-              .headers(
-                  httpHeaders ->
-                      OpenTelemetry.getPropagators()
-                          .getHttpTextFormat()
-                          .inject(Context.current(), httpHeaders, SETTER))
+              .headers(httpHeaders -> DECORATE.inject(Context.current(), httpHeaders))
               .build();
       DECORATE.onRequest(span, mutatedRequest);
 
       return next.exchange(mutatedRequest)
           .doOnSuccessOrError(
               (clientResponse, throwable) -> {
-                if (throwable != null) {
-                  DECORATE.onError(span, throwable);
-                } else {
-                  DECORATE.onResponse(span, clientResponse);
-                }
+                DECORATE.onError(span, throwable);
```

Comment:
There's a `throwable != null` check in DECORATE.onError(). This if statement seems redundant.

Reply:
I think we have some deep checks like that but would prefer code to read better, rather than be minimum LoC - I'm thinking some day such a deep null check will be gone.

Here, it doesn't really sound right to call `onError` when it's not an error, so having the if check makes this code more readable than without it IMO (and if I understood him correctly, @iNikem too :) )

Reply:
Decorators usually have null checks inside methods. New Tracers usually has them outside.

But in this particular case, you have to decide either call this method or that method. Calling `onError` always assuming/hoping that it is a no-op if `throwable` is null does not seem write to me.